### PR TITLE
[CDAP-6349] - Fixes number widget to honor max & min values from widget json

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Fixes querying max & min values from widget json
- Fixes 0 (being a falsy condition) to be converted to string as backend only accepts strings today.

Bamboo build: http://builds.cask.co/browse/CDAP-DRC3948-1

### CODE REVIEW

**Comment by ajainarayanan** [comment]
##### EDIT:
- Fixes the min & max to be strings 
- Fixes `timeline` to be part of es6Directives in Gulpfile.js
- Fixes scrolling in config modal in safari to be hidden (had two scrollbars inside config modal).

##### Caveat:
- The user can still enter a negative number even if the min value is 0, but the negative number is not stored in the model (pipeline config as part of the stage).
- There is also this case where we check for empty value & if it is empty we use the default value. So now if the user erases the existing value and close the modal & reopens again we populate it with default value as we use default value if none present as part of the widget.

File: cdap-ui/app/features/hydratorplusplus/hydrator-modal.less [hunk 1]
```diff
@@ -148,7 +148,7 @@
 
         .modal-body {
           padding: 0 10px;
-          overflow: auto;
+          overflow: hidden;
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/hydrator-modal.less:152]: this does not cause problem when the config is long?

Comment (ajainarayanan) [inline cdap-ui/app/features/hydratorplusplus/hydrator-modal.less:152]: Yeah. For some reason safari shows two levels of scroll bars one for the entire node config modal and the other for the tab contents. :|

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by astewes** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
Talked to @nitinmotgi This fix would suffice now. Creating another JIRA for 4.0 to fix this issue by showing a validation error as and when the user types in [JIRA - CDAP-6522](https://issues.cask.co/browse/CDAP-6522)

**Pull request merged**